dasharo-security/secure-boot.robot

```diff
@@ -129,281 +121,210 @@ SBO003.001 Attempt to boot file with the correct key from Shell (firmware)
     ...    -out DB-ECDSA384.crt -days 3650 -nodes -sha256`
     ...    converted to DER using `openssl base64 -d -in DB-ECDSA384.cer -out DB-ECDSA384.der`
-    Mount ISO As USB In QEMU    ${CURDIR}/../scripts/secure-boot/images/ECDSA384.img
+    Skip If    not ${SECURE_BOOT_SUPPORT}    SBO010.005 not supported
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SBO010.005 not supported
+    Mount ISO As USB    ${CURDIR}/../scripts/secure-boot/images/ECDSA384.img
     Power On
 
     ${sb_menu}=    Enter Secure Boot Menu And Return Construction
     Enable Secure Boot    ${sb_menu}
-    ${sb_menu}=    Reenter Menu And Return Construction
-    ${advanced_menu}=    Enter Advanced Secure Boot Keys Management And Return Construction    ${sb_menu}
-    Enter Enroll DB Signature Using File In DB Options    ${advanced_menu}
-    Enter Volume In File Explorer    ECDSA384
-    Select File In File Explorer    cert.der
-    Save Changes And Reset    3    5
+    ${key_menu}=    Enter Key Management And Return Construction
+    Enroll DB Signature    ${key_menu}    ECDSA384    cert.der
+    Save Changes And Reset
 
-    Enter UEFI Shell
-    ${out}=    Execute File In UEFI Shell    signed-hello.efi
-    Should Contain    ${out}    Hello, world!
+    Boot Efi File    signed-hello.efi    ECDSA384    Hello, world!
 
 SBO010.006 Check support for ecdsa521 signed certificates
     [Documentation]    PEM generated with `openssl req -new -x509 -newkey ec -pkeyopt ec_paramgen_curve:P-521 -subj "/CN=DB-ECDSA521/" -keyout DB-ECDSA521.key
     ...    -out DB-ECDSA521.crt -days 3650 -nodes -sha256`
     ...    converted to DER using `openssl base64 -d -in DB-ECDSA521.cer -out DB-ECDSA521.der`
-    Mount ISO As USB In QEMU    ${CURDIR}/../scripts/secure-boot/images/ECDSA521.img
+    Skip If    not ${SECURE_BOOT_SUPPORT}    SBO010.006 not supported
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SBO010.006 not supported
+    Mount ISO As USB    ${CURDIR}/../scripts/secure-boot/images/ECDSA521.img
     Power On
 
     ${sb_menu}=    Enter Secure Boot Menu And Return Construction
     Enable Secure Boot    ${sb_menu}
-    ${sb_menu}=    Reenter Menu And Return Construction
-    ${advanced_menu}=    Enter Advanced Secure Boot Keys Management And Return Construction    ${sb_menu}
-    Enter Enroll DB Signature Using File In DB Options    ${advanced_menu}
-    Enter Volume In File Explorer    ECDSA521
-    Select File In File Explorer    cert.der
-    Save Changes And Reset    3    5
+    ${key_menu}=    Enter Key Management And Return Construction
+    Enroll DB Signature    ${key_menu}    ECDSA521    cert.der
+    Save Changes And Reset
 
-    Enter UEFI Shell
-    ${out}=    Execute File In UEFI Shell    signed-hello.efi
-    Should Contain    ${out}    Hello, world!
+    Boot Efi File    signed-hello.efi    ECDSA521    Hello, world!
 
 SBO011.001 Attempt to enroll expired certificate and boot signed image
```

Comment:
@m-iwanicki Faced following error while running tests on commit `236b638a06073132378566a98d62c37195c7c83e`:

```bash
(...)
SBO011.001 Attempt to enroll expired certificate and boot signed i... | FAIL |
No match found for 'exit' in 30 seconds. Output:
------------------------------------------------------------------------------
(...)
```

Reply:
@TomaszAIR Referencing to [your comment](https://github.com/Dasharo/open-source-firmware-validation/pull/194#discussion_r1466302339). What to do with this test then? I need to prepare manual steps for this steps in `Dasharo/docs`  for Qemu using Dasharo firmware. Should I include this test to documentation or not?

Reply:
The test has been added to `Dasharo/docs`.